--- ./nginx_ajp_module/ngx_http_ajp.c   2023-02-27 12:53:45.156017414 +0900
+++ ./nginx_ajp_module/ngx_http_ajp.c.new   2023-02-27 13:01:40.856841005 +0900
@@ -136,7 +136,7 @@
 
     escape = 0;
 
-    if (r->quoted_uri || r->space_in_uri || r->internal) {
+    if (r->quoted_uri || r->internal) {
         escape = 2 * ngx_escape_uri(NULL, r->uri.data,
                 r->uri.len, NGX_ESCAPE_URI);
     }
